src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -441,6 +444,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addSuperinterface(superinterface: KClass<*>)
         = addSuperinterface(superinterface.asTypeName())
 
+    fun addSuperinterface(superinterface: ParameterSpec) = apply {
```

Comment:
delegate can be a `val` or simply a dummy parameter that's not doing anything.  I'm all for keeping it simple, but there's always going to be an edge case... Modifiers on the input argument too? It's a bit more complicated because it's affecting multiple aspects of the type